feat: Added global reach peering connection to express-route-circuit module

arnoldna opened the pull request:
## Description

- Added global reach peering connection to express-route-circuit module
- Added global-reach test
- Updated API version `Microsoft.Network/expressRouteCircuits@2024-10-01`

Closes #6108 

## Pipeline Reference

<!-- Insert your Pipeline Status Badge below -->

| Pipeline |
| -------- |
[![avm.res.network.express-route-circuit](https://github.com/arnoldna/bicep-registry-modules/actions/workflows/avm.res.network.express-route-circuit.yml/badge.svg?branch=avm%2Fres%2Fnetwork%2Fer)](https://github.com/arnoldna/bicep-registry-modules/actions/workflows/avm.res.network.express-route-circuit.yml)

## Type of Change

<!-- Use the checkboxes [x] on the options that are relevant. -->

- Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
  - [X] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [X] Update to documentation
- [ ] Update to CI Environment or utilities (Non-module affecting changes)

## Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] I have run `Set-AVMModule` locally to generate the supporting module files.
- [X] My corresponding pipelines / checks run clean and green without any errors or warnings
- [X] I have updated the module's CHANGELOG.md file with an entry for the next version

<!--  Please keep up to date with the contribution guide at https://aka.ms/avm/contribute/bicep -->

arnoldna committed:
Added support for Express Route Ports

arnoldna committed:
Merge branch 'main' of https://github.com/arnoldna/bicep-registry-modules into avm/res/network/er

arnoldna committed:
Updated documentation and version

arnoldna committed:
Doc updates

arnoldna committed:
Updated docs

arnoldna committed:
Merge branch 'main' of https://github.com/arnoldna/bicep-registry-modules into avm/res/network/er

arnoldna committed:
AVM-RES: Update ExpressRoute Circuit module and tests

- Refactored peering parameters to use an array of configurations for better flexibility.
- Updated resource API versions for ExpressRoute Circuit and resource group.
- Added route filter resource to dependencies for enhanced functionality.
- Adjusted test parameters to reflect new configurations and ensure compatibility.

arnoldna committed:
Update Express Route Circuit module to version 0.7.0

- Updated API version for `Microsoft.Network/expressRouteCircuits` to 2024-07-01.
- Changed example `name` parameter to use a prefix placeholder.
- Added support for `globalReachEnabled` parameter.
- Introduced detailed peering configurations including IPv6 settings.
- Adjusted test files to align with new parameter structures.

arnoldna committed:
Merge branch 'main' of https://github.com/arnoldna/bicep-registry-modules into avm/res/network/er

arnoldna committed:
AVM-RES: Update namePrefix parameter in main.test.bicep

- Changed default value of namePrefix parameter from 'nga' to '#_namePrefix_#' for consistency.
- Ensures that the parameter can be dynamically set during deployment.

arnoldna committed:
AVM-RES: Update version and changelog for express-route-circuit

- Updated version to 0.6 in version.json
- Revised changelog to reflect changes for version 0.6.1
- Added details for LockType update and API version change

arnoldna committed:
AVM-RES: Update name parameters in README and tests

- Changed name parameter from '#_namePrefix_#001' to 'nercmax001' in README.md
- Updated serviceShort parameter from '#_namePrefix_#' to 'nercmax' in main.test.bicep
- Ensures consistency in naming conventions across module and tests

arnoldna committed:
AVM-RES: Fix typos in README and output descriptions

- Corrected spelling of "circuit" in README and output descriptions in main.bicep and main.json files.
- Ensured consistency in terminology across documentation and code outputs for the express route circuit module located at avm/res/network/express-route-circuit.

arnoldna committed:
AVM-RES: Update IPv6 peering configuration in module

- Added `routeFilter` parameter to IPv6 peering configuration in README and Bicep files.
- Introduced `advertisedPublicPrefixInfo` parameter for IPv6 prefix information in Bicep and JSON definitions.
- Updated state handling in Bicep to use the generic `state` parameter instead of `ipv4State`.
- Removed deprecated `ipv6PeeringConfigType` from JSON definitions.

arnoldna committed:
Added Resource Defined Types for "Peerings".

arnoldna committed:
Merge branch 'main' into avm/res/network/er

arnoldna committed:
AVM-RES: Update version to 0.7 in version.json

- Updated version number in version.json from 0.6 to 0.7.
- Updated changelog to reflect changes in version 0.6.1.

arnoldna committed:
Merge branch 'avm/res/network/er' of https://github.com/arnoldna/bicep-registry-modules into avm/res/network/er

arnoldna committed:
AVM-RES: Update changelog and main.json for ExpressRoute

- Updated changelog to version 0.7.0, reflecting new features.
- Removed outdated entries from changelog for clarity.
- Updated metadata version and template hash in main.json.

arnoldna committed:
Merge branch 'main' into avm/res/network/er

arnoldna committed:
Merge branch 'main' of https://github.com/arnoldna/bicep-registry-modules into avm/res/network/er

arnoldna committed:
Merge branch 'avm/res/network/er' of https://github.com/arnoldna/bicep-registry-modules into avm/res/network/er

arnoldna committed:
AVM-RES: Enhance ExpressRoute Circuit module

- Added optional parameter for Global Reach connection configurations.
- Updated ExpressRoute circuit resource API version to 2024-10-01.
- Introduced new test dependencies for Global Reach connections.
- Created test cases for deploying ExpressRoute circuits with Global Reach.

arnoldna committed:
AVM-RES: Update ExpressRoute Circuit module

- Updated API version for ExpressRoute circuits to 2024-10-01.
- Added support for Global Reach connections in the module.
- Enhanced README with new usage examples and parameters.
- Added output for authorizations of the ExpressRoute circuit.
- Updated dependencies for testing Global Reach connections.

arnoldna committed:
Merge branch 'main' of https://github.com/arnoldna/bicep-registry-modules into avm/res/network/er

arnoldna committed:
AVM-RES: Update changelog and test module name

- Added changelog entry for version 0.7.1 with Global Reach peerings support.
- Updated test module name from `testDeploymentCircuit2` to `testDeployment` for clarity.

arnoldna committed:
AVM-RES: Remove outdated test files for Global Reach

- Deleted `dependencies.bicep` and `main.test.bicep` test files for the ExpressRoute Circuit module.
- These files were no longer necessary for the current testing framework.

arnoldna committed:
AVM-RES: Update changelog and version for ExpressRoute Circuit

- Updated changelog to version 0.8.0
- Added support for Global Reach peerings connections
- Updated API version `Microsoft.Network/expressRouteCircuits@2024-10-01`
- Updated version.json to reflect version 0.8

arnoldna committed:
Merge branch 'main' into avm/res/network/er

arnoldna committed:
Merge branch 'main' into avm/res/network/er

AlexanderSehr commented on avm/res/network/express-route-circuit/tests/e2e/global-reach/dependencies.bicep:
```diff
@@ -0,0 +1,90 @@
+}
+
+// Deploy second circuit
+module testDeploymentCircuit1 '../../../main.bicep' = {
```
Dependencies must not use the module itself. Please either use a previously published version or the plain `resource`.
Dependencies should be 'stable' and not break upon changing anything in the module.

arnoldna replied:
Done.

arnoldna committed:
AVM-RES: update module reference for testDeploymentCircuit1

- Changed module reference for testDeploymentCircuit1 to use the correct path.
- Ensures the deployment uses the intended Express Route Circuit module.
